### PR TITLE
Evolution Endpoint

### PR DESCRIPTION
**Pull request by WesselBBD**
...

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -59,7 +59,7 @@
 		<dependency>
 			<groupId>org.projectlombok</groupId>
 			<artifactId>lombok</artifactId>
-			<optional>true</optional>
+			<version>1.18.24</version>
```
[the remark this answers is not in the document]
Comment (WesselBBD) [reply pom.xml:62]: This is lombok's version and should be [fine](https://projectlombok.org/changelog). Regardless it's working with Corretto 17 in the live environment which is good enough for me

File: src/main/java/com/pokedex/pokeAPI/repositories/PokemonSpeciesRepository.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+public interface PokemonSpeciesRepository extends CrudRepository<PokemonSpeciesData, Integer> {
+    @Query("SELECT psd FROM PokemonSpeciesData psd WHERE psd.evolution_chain_id = :fam")
+    List<PokemonSpeciesData> getFamily(@Param("fam") Integer famId);
```
[the remark this answers is not in the document]
Comment (WesselBBD) [reply src/main/java/com/pokedex/pokeAPI/repositories/PokemonSpeciesRepository.java:15]: The drived one didn't work for some reason. I'll try again

File: src/main/java/com/pokedex/pokeAPI/services/EvolutionService.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        if (request.size() == 0) return Collections.emptyList();
+
+        var requestPokemon = request.get(0);
+
+        var requestPokemonS = requestPokemon.getPokemon_species();
+        
+        var requestPokemonFamily = pokemonSpeciesRepo.getFamily(requestPokemonS.getEvolution_chain_id());
+
+        requestPokemonFamily.sort((sp1, sp2) -> sp1.getOrder().compareTo(sp2.getOrder()));
```
[the remark this answers is not in the document]
Comment (WesselBBD) [reply src/main/java/com/pokedex/pokeAPI/services/EvolutionService.java:41]: Fair

**Review by ChadleyM — APPROVED** [review]
Look good

**Pull request merged**